### PR TITLE
Getting VAD states from the riva-server

### PR DESCRIPTION
**Pull request by sarane22**
Adding support to get VAD probabilities for the riva server.

### CODE REVIEW

File: riva/client/asr.py [hunk 1]
```diff
@@ -248,6 +248,12 @@ def print_streaming(
                 continue
             partial_transcript = ""
             for result in response.results:
+                if result.pipeline_states and len(result.pipeline_states.vad_probabilities) > 0:
```
[the remark this answers is not in the document]
Comment (sarane22) [reply riva/client/asr.py:251]: Yes, whenever the client sends --custom_configuration="get_vad_probabilities:true" it'll get logged in the file.

File: riva/client/asr.py [hunk 2]
```diff
@@ -248,6 +248,12 @@ def print_streaming(
...
+                    vad_prob_logs = "VAD States: "
+                    for vad_state in result.pipeline_states.vad_probabilities:
+                            vad_prob_logs += str(vad_state) + " "
+                    for i, f in enumerate(output_file):
```
Comment (atomer-nvidia) [inline riva/client/asr.py:255]: Are we not creating a new output file for dumping vad probabilities like in cpp client? Would that interfere with actual transcripts getting dumped into file? If both custom configuration is set to print vad probabilities and also print transcript flag is set to true, how does the content of output file look like?

File: riva/client/asr.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sarane22) [reply riva/client/asr.py:251]: yes, here: https://docs.nvidia.com/deeplearning/riva/user-guide/docs/reference/protos/protos.html?highlight=custom_configuration

[the remark this answers is not in the document]
Comment (atomer-nvidia) [reply riva/client/asr.py:251]: Sorry, I meant is "get_vad_probabilities" documented somewhere that it can be passed in custom_configuration?

**Review by atomer-nvidia — APPROVED** [review]

File: common
Comment (atomer-nvidia) [inline common:1]: This needs to be updated to latest TOT common.

**Comment by menon92** [comment]
Hi @sarane22 thanks for adding this feature in Riva. 

I'm trying to pring VAD state with the Riva [marblenet-vad](https://docs.nvidia.com/deeplearning/riva/user-guide/docs/asr/asr-pipeline-configuration.html#marblenet-vad) but i'm getting no value under `result.pipeline_states`

I'm using riva 2.18.0. Do you have any clue on it?

Thanks

**Pull request merged**